contrib/android/build.sh

```diff
@@ -26,7 +26,8 @@ cp "$CONTRIB/deterministic-build/requirements-build-android.txt" "$CONTRIB_ANDRO
 sudo docker build \
     $DOCKER_BUILD_FLAGS \
     -t electrum-android-builder-img \
-    "$CONTRIB_ANDROID"
+    --file "$CONTRIB_ANDROID/Dockerfile" \
+    "$PROJECT_ROOT"
 rm "$CONTRIB_ANDROID/requirements-build-android.txt"
```

Comment:
I guess with this change the `cp` and `rm` lines are not needed anymore?

Reply:
> I guess with this change the `cp` and `rm` lines are not needed anymore?

@SomberNight I think that's correct.



> Note that a side-effect of this change is sending a potentially much bigger build context to docker. E.g. consider having build caches for all the different builds on your local machine; all that would now be included.
> Couldn't we use instead the same `cp` approach in the cirrus yml?

The issue here is that Cirrus's automatic Dockerfile-to-container-image generator only supports using the repo root as the `PATH` argument to `docker build`.  Or at least, if Cirrus supports custom `PATH` arguments, it isn't documented.  I'll check with Cirrus support and see if there's a good workaround for this.

Reply:
Okay, so, Cirrus implemented support for custom Docker contexts, and it works.  However, there is another hitch: Cirrus does not support running custom scripts before it builds the Docker container from the Dockerfile.  So, we can't run `cp` to copy `requirements-build-android.txt` from `contrib/deterministic-build` to `contrib/android`.  Symlinks do not work as a workaround.  I see a few options:

1. Use `contrib/deterministic-build` as the build context, since the only file being copied into the container is in that directory.  Might constrain future changes to the Dockerfile.
2. Use `contrib` as the build context, since that includes both `android` and `deterministic-build`.  Bigger context to cache than `contrib/android`, but better than the repo root.
3. Manually check the results of `cp` into Git every time `requirements-build-android.txt` is updated.  Seems error-prone and a pain to audit.
4. Ask Cirrus to support custom hook scripts in their automatic docker_builder VM's.

My preference is to do (1) as a short-term solution, and pursue (4) as a longer-term solution that doesn't block this PR.  @SomberNight does that approach work for you?

Reply:
> 1. Use `contrib/deterministic-build` as the build context, since the only file being copied into the container is in that directory.  Might constrain future changes to the Dockerfile.

Does this work? My understanding was that the Dockerfile must be located somewhere inside the build context.
This is also stated [in the docs](https://docs.docker.com/engine/reference/commandline/build/#text-files):
> By default the docker build command will look for a Dockerfile at the root of the build context. The -f, --file, option lets you specify the path to an alternative file to use instead. This is useful in cases where the same set of files are used for multiple builds. The path must be to a file within the build context.

Reply:
Ok... how about we do `ln -s .gitignore .dockerignore`, commit `.dockerignore` into the repo, and use `$PROJECT_ROOT` as the build context for the android build?

Reply:
> Does this work? My understanding was that the Dockerfile must be located somewhere inside the build context.

It definitely looks like it worked, see https://github.com/JeremyRand/electrum-nmc/commit/9f224ae1cc3acf41e94f226e28a8ba3540a31406 (the Android Docker build on Cirrus succeeded for that commit).  I don't know if maybe that's relying on undefined behavior and I just got lucky though.  :/

> Ok... how about we do `ln -s .gitignore .dockerignore`, commit `.dockerignore` into the repo, and use `$PROJECT_ROOT` as the build context for the android build?
> (related: [126afeb](https://github.com/spesmilo/electrum/commit/126afebea1cd774d6f610dbefe84348ed9f278a3))

Ah, that's a clever trick.  I have no objection to this if you're okay with it.  I will make that change and verify that it still works, and then update this PR if all looks good.